src/ripple/app/main/Application.cpp

```diff
@@ -1489,7 +1498,9 @@ ApplicationImp::signalStop()
 {
     // Unblock the main thread (which is sitting in run()).
     //
-    m_stop.signal();
+    std::lock_guard<std::mutex> lk{mut_};
+    time_to_stop = true;
+    cv_.notify_all();
```

Comment:
Consider narrowing the lock scope as it should not be necessary to notify.

eg.
```
{
    std::lock_guard<std::mutex> lk{mut_};
    time_to_stop = true;
}
cv_.notify_all();
```